src/main/java/io/github/jhipster/config/apidoc/SwaggerAutoConfiguration.java

```diff
@@ -77,74 +94,91 @@ public SwaggerConfiguration(JHipsterProperties jHipsterProperties) {
      * @return the Swagger Springfox configuration
      */
     @Bean
-    public Docket swaggerSpringfoxApiDocket() {
+    @ConditionalOnMissingBean(name = "swaggerSpringfoxApiDocket")
+    public Docket swaggerSpringfoxApiDocket(List<SwaggerCustomizer> swaggerCustomizers,
+                                            List<AlternateTypeRule> alternateTypeRules) {
         log.debug(STARTING_MESSAGE);
         StopWatch watch = new StopWatch();
         watch.start();
-        Contact contact = new Contact(
-            properties.getContactName(),
-            properties.getContactUrl(),
-            properties.getContactEmail());
 
-        ApiInfo apiInfo = new ApiInfo(
-            properties.getTitle(),
-            properties.getDescription(),
-            properties.getVersion(),
-            properties.getTermsOfServiceUrl(),
-            contact,
-            properties.getLicense(),
-            properties.getLicenseUrl(),
-            new ArrayList<>());
+        Docket docket = createDocket();
+
+        // Apply all SwaggerCustomizers.
+        swaggerCustomizers.forEach(customizer -> customizer.customize(docket));
+
+        // Add all AlternateTypeRules.
+        // Also you can add you rules in a customizer bean above.
+        docket.alternateTypeRules(alternateTypeRules.toArray(new AlternateTypeRule[alternateTypeRules.size()]));
 
-        Docket docket = createDocket()
-            .host(properties.getHost())
-            .protocols(new HashSet<>(Arrays.asList(properties.getProtocols())))
-            .apiInfo(apiInfo)
-            .forCodeGeneration(true)
-            .directModelSubstitute(ByteBuffer.class, String.class)
-            .genericModelSubstitutes(ResponseEntity.class)
-            .select()
-            .paths(regex(properties.getDefaultIncludePattern()))
-            .build();
         watch.stop();
         log.debug(STARTED_MESSAGE, watch.getTotalTimeMillis());
         return docket;
     }
 
+    /**
+     * JHipster Swagger Customizer
+     *
+     * @return the Swagger Customizer of JHipster
+     */
+    @Bean
+    public JHipsterSwaggerCustomizer jHipsterSwaggerCustomizer() {
+        return new JHipsterSwaggerCustomizer(properties);
+    }
+
+    @Bean
+    public AlternateTypeRule responseEntityAlternateTypeRule(TypeResolver typeResolver) {
+        return AlternateTypeRules.newRule(
```

Comment:
Why changing it to AlternateTypeRules ? It seemed easier to have it in the docket customizer (only one line of code) !

Reply:
Inject directly a List of AlternateTypeRules will cause an exception if there is no  AlternateTypeRule bean. I already did such trial in a demo project.  So I have to hold at least one AlternateTypeRule bean. Or may I change it back to `ObjectProvider`? I would like to hear your advice. :-)

Reply:
Oh, I wasn't aware of that ! Then the `ObjectProvider` way seems the way to go ! Sorry for the wrong comment and all the changes ...

Reply:
That's OK. Fixed.